server/src/main/java/org/elasticsearch/index/search/stats/FieldUsageStats.java

```diff
@@ -0,0 +1,279 @@
+            out.writeVLong(storedFields);
+            out.writeVLong(norms);
+            out.writeVLong(payloads);
+            out.writeVLong(termVectors);
+            out.writeVLong(points);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(TOTAL, getTotal());
+            builder.field(TERMS, terms);
+            builder.field(PROXIMITY, getProximity());
+            builder.field(STORED_FIELDS, storedFields);
+            builder.field(DOC_VALUES, docValues);
+            builder.field(POINTS, points);
+            builder.field(NORMS, norms);
+            builder.field(TERM_VECTORS, termVectors);
+            builder.field(FREQS, freqs);
+            builder.field(POSITIONS, positions);
+            builder.field(OFFSETS, offsets);
+            builder.field(PAYLOADS, payloads);
+            return builder;
+        }
+
+        public Set<UsageContext> keySet() {
+            final EnumSet<UsageContext> set = EnumSet.noneOf(UsageContext.class);
+            if (terms > 0L) {
+                set.add(UsageContext.TERMS);
+            }
+            if (freqs > 0L) {
+                set.add(UsageContext.FREQS);
+            }
+            if (positions > 0L) {
+                set.add(UsageContext.POSITIONS);
+            }
+            if (offsets > 0L) {
+                set.add(UsageContext.OFFSETS);
+            }
+            if (docValues > 0L) {
+                set.add(UsageContext.DOC_VALUES);
+            }
+            if (storedFields > 0L) {
+                set.add(UsageContext.STORED_FIELDS);
+            }
+            if (norms > 0L) {
+                set.add(UsageContext.NORMS);
+            }
+            if (payloads > 0L) {
+                set.add(UsageContext.PAYLOADS);
+            }
+            if (termVectors > 0L) {
+                set.add(UsageContext.TERM_VECTORS);
+            }
+            if (points > 0L) {
+                set.add(UsageContext.POINTS);
+            }
+            return set;
+        }
+
+        public long getTerms() {
+            return terms;
+        }
+
+        public long getFreqs() {
+            return freqs;
+        }
+
+        public long getPositions() {
+            return positions;
+        }
+
+        public long getOffsets() {
+            return offsets;
+        }
+
+        public long getDocValues() {
+            return docValues;
+        }
+
+        public long getStoredFields() {
+            return storedFields;
+        }
+
+        public long getNorms() {
+            return norms;
+        }
+
+        public long getPayloads() {
+            return payloads;
+        }
+
+        public long getTermVectors() {
+            return termVectors;
+        }
+
+        public long getPoints() {
+            return points;
+        }
+
+        public long getProximity() {
+            return freqs + offsets + positions + payloads;
+        }
```

Comment:
FYI `freqs` is not considered part of proximity in @dnhatn 's PR about disk usage. Since I believe that you only added proximity here so that users could more easily relate the output of this API to the disk usage one, let's remove freqs from here?

Also could we track proximity separately instead of summing up other counters so that it would increment by only one per request even if both positions and offsets are used?

Reply:
> FYI freqs is not considered part of proximity in @dnhatn 's PR about disk usage. Since I believe that you only added proximity here so that users could more easily relate the output of this API to the disk usage one, let's remove freqs from here?

Good catch! I think the terminology is becoming problematic in @dnhatn's PR as well then. What he calls "terms" includes postings + frequencies.

Reply:
Agreed, maybe this other PR should call it `inverted_index` or something like that.